src/twisted/internet/defer.py

```diff
@@ -52,6 +52,12 @@ class TimeoutError(Exception):
     """
 
 
+class InvalidStateError(Exception):
```

Comment:
This should maybe derive from ``BaseException`` instead so that a:

```python
try:
    undefer(d)
except Exception:
    pass

```

Does not catch an invalid ``undefer()`` call the same as an exception that came from the callbacks internal to it.